client/consul.go

```diff
@@ -26,6 +36,43 @@ func NewConsulAPI() *APIClient {
 	return NewAPIClient("consul", addr, headers)
 }
 
+// NewConsulTLSConfig returns a *TLSConfig object, using
+// default environment variables to build up the object.
+func NewConsulTLSConfig() (*TLSConfig, error) {
+	tlsConfig := TLSConfig{}
+	if v := os.Getenv(EnvConsulCaCert); v != "" {
+		tlsConfig.CACert = v
+	}
```

Comment:
I have made an update to skip the conditional checks; it's a good call, since the default will be exactly the same as if the check wasn't there. I did it slightly differently, by just initializing tlsConfig with the fields set with environment variables, rather than creating the object and then cycling through fields. Of course, the verification piece is a bit different.

Reply:
If it's all right with you, @gulducat , I think I'm going to skip the map for the time being. For some reason, I feel like if I went down that path, I would also want to not have to repeat, for example "CACert" in `tlsConfig.CACert = os.Getenv(env["CACert"])`. I'd like the name of the field to be automagically determined. But, I don't know if there's a good way to do that without reflection, which feels like it could be a rabbit hole of its own and could ultimately be less readable.

Reply:
works for me!  I do still think there may be value in it, but it seems much less important now with [33d0389](https://github.com/hashicorp/hcdiag/pull/99/commits/33d03892b88c82b994a807b8525950a812aa8b5c) making each product's config function a lot tidier (for me) to skim at a glance.  👍